lib/movable_ink/aws/ec2.rb

```diff
@@ -91,18 +91,26 @@ def me
         @me ||= all_instances.select{|instance| instance.instance_id == instance_id}.first rescue nil
       end
 
-      def instances(role:, exclude_roles: [], region: my_region, availability_zone: nil, exact_match: false)
-        instances = all_instances(region: region).select { |instance|
-          instance.tags.select{ |tag| tag.key == 'mi:roles' }.detect { |tag|
-            roles = tag.value.split(/\s*,\s*/)
-            if exact_match
-              roles == [role]
-            else
-              exclude_roles.push('decommissioned')
-              roles.include?(role) && !roles.any? { |role| exclude_roles.include?(role) }
-            end
+      def instances(role:, exclude_roles: [], region: my_region, availability_zone: nil, exact_match: false, use_cache: true)
+        if exact_match && use_cache == false
+          filter = default_filter.push({
+            name: 'tag:mi:roles',
+            values: [role]
+          })
+          instances = load_all_instances(my_region, filter: filter)
```

Comment:
@aackerman your example should work fine.  If you don't explicitly set `use_cache` to `false`, the behavior remains the same as it's always been and the API will still be queried for all instances in your environment.  If you include `use_cache: false`, it will still work as desired, but you will make two filtered API calls rather than one unfiltered API call and the results of those two calls would not be cached within the library.